docs/data/toolpad/building-ui/datagrid-component.md

```diff
@@ -0,0 +1,38 @@
+# DataGrid component
+
+<p class="description">Build powerful and feature rich data tables</p>
+
+<img src="/static/toolpad/docs/building-ui/datagrid.png" width="884px" alt="Custom component" />
+
+## Customizing cell rendering
+
+By default you can render json data using DataGrid component and it will automatically recognize most of the data types. However sometimes it might not be enough and for that reason we provide the option to render custom components inside a datagrid.
+
+### Creating custom component
+
+Firstly, you will have to create a custom component which should be able to process `DataGrid` data:
+
+- `value` prop will receive the value of the specific column
+- `row` prop will receive the data for the entire row
+- `field` prop will receive field configuration data
+
+<img src="/static/toolpad/docs/building-ui/cell-component.png" width="621px" alt="Custom component" />
+
+### Configuring columns
+
```

Comment:
Perhaps it could be interesting to have a section for each column type?

<img width="286" alt="Screenshot 2023-02-02 at 11 40 57" src="https://user-images.githubusercontent.com/2109932/216303009-2c63592b-1c42-47fb-ae61-eb1892f1059e.png">